app/models/user.rb

```diff
@@ -3,13 +3,30 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable,
-         :omniauthable, omniauth_providers: [:twitter]
+         :omniauthable, omniauth_providers: [:twitter],
+         :authentication_keys => [:login]
+
   before_destroy :check_all_events_finished
 
   has_many :created_events, class_name: 'Event', foreign_key: :owner_id, dependent: :nullify
   has_many :tickets, dependent: :nullify
   has_many :participating_events, through: :tickets, source: :event
 
+  attr_accessor :login
+
+  def self.find_first_by_auth_conditions(warden_conditions)
+    conditions = warden_conditions.dup
+    if login = conditions.delete(:login)
+      where(conditions).where(["username = :value OR lower(email) = lower(:value)", { :value => login }]).first
```

Comment:
__[rubocop]__
* Rails/FindBy: Use `find_by` instead of `where.first`.
* Style/HashSyntax: Use the new Ruby 1.9 hash syntax. [:ref](https://github.com/bbatsov/ruby-style-guide#hash-literals)
* Metrics/LineLength: Line is too long. [111/100] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)